### PR TITLE
[rdc/scn] Revise reset scenario

### PR DESCRIPTION
**Pull request by eunchan**
Revise aon Power OK reset scenario.

- vcaon_pok reset signal from the regulator
- toggle vcmain_pok together (as expected scenario)

Now 2003 errors and 307 warnings.

### CODE REVIEW

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl [hunk 1]
```diff
@@ -14,36 +14,87 @@
...
   { POR_N           { reset {#2 0} { #10 1} }} \
-  { u_ast.vcaon_pok { constraint {@t0 1} }} \
+  { u_ast.u_rglts_pdm_3p3v.vcaon_pok_h  { constraint {@t0 1} }} \
   { top_earlgrey.u_pinmux_aon.dio_pad_attr_q[12].invert \
     { constraint { @t0 0 } } } \
   { top_earlgrey.u_spi_device.cio_sck_i { constraint { @t0 0 } } } \
 } -name ScnPOR
 
-# AST POK
+# AST AON POK
```
Comment (msfschaffner) [inline hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:28]: I will have to take a closer look at the ones below - will comment in a bit.

Comment (msfschaffner) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:28]: The scenarios LGTM, although I was wondering whether we should ask Nuvoton to look these over also? In essence, we are capturing AST behavior here, right?

@tjaychen WDYT?

Comment (tjaychen) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:28]: yeah @eunchan and i were discussing earlier, it looks good to me.
Maybe we can just confirm with Nuvoton / Ziv via email? Probably don't need them to look through all of this...

File: hw/top_earlgrey/rdc/rdc_waivers.tcl [hunk 2]
```diff
@@ -19,3 +19,15 @@ foreach mod $modules {
 set_rule_status -rule {S_GENCLK} -status {Waived}  \
   -expression {(ClockTreeSignal=~"IOC6")} \
   -comment {External signal has SW bypass option. But usually not used. So internal generated clock is define ans main clock}
+
+# RstMgrSwRst scenario does not involved in retention logic
```
[the remark this answers is not in the document]
Comment (eunchan) [reply hw/top_earlgrey/rdc/rdc_waivers.tcl:23]: Yes.

File: hw/top_earlgrey/rdc/rdc_waivers.spi_device.tcl [hunk 3]
```diff
@@ -128,3 +128,7 @@ set_rule_status -rule {E_RST_METASTABILITY} -status {Waived} \
   -expression {(ResetFlop=~"*u_spi_device.u_reg.u_addr_swap_mask.q*") && \
     (MetaStableFlop=~"*u_pinmux_aon.dio_out_ret*") } \
   -comment {When SwRst issued, chip is in active mode.}
+set_rule_status -rule {E_RST_METASTABILITY} -status {Waived} \
+  -expression {(ResetFlop=~"*u_spi_device.u_reg.u_addr_swap_mask.q*") && \
+    (MetaStableFlop=~"*u_pinmux_aon.dio_out*")} \
```
[the remark this answers is not in the document]
Comment (eunchan) [reply hw/top_earlgrey/rdc/rdc_waivers.spi_device.tcl:133]: Hum. Maybe I waived this first then blindly waived to any retention regs. Let me remove this.

[the remark this answers is not in the document]
Comment (eunchan) [reply hw/top_earlgrey/rdc/rdc_waivers.spi_device.tcl]: Removed.

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl [hunk 4]
```diff
@@ -19,21 +19,24 @@ set_reset_scenario { \
   { u_ast.vcaon_pok { constraint {@t0 1} }} \
   { top_earlgrey.u_pinmux_aon.dio_pad_attr_q[12].invert \
     { constraint { @t0 0 } } } \
-  { top_earlgrey.u_spi_device.cio_sck_i { constraint { @t0 S } } } \
+  { top_earlgrey.u_spi_device.cio_sck_i { constraint { @t0 0 } } } \
```
[the remark this answers is not in the document]
Comment (eunchan) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:25]: Even the issues are addressed, still I am sure this line needed due to the simplicity of the RDC tool.

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl (hunk 1 above)
Comment (eunchan) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:28]: Yep. I added this based on Tim's suggestion. Thanks Tim!

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl (hunk 4 above)
[the remark this answers is not in the document]
Comment (msfschaffner) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:25]: Ok got it.

**Review by msfschaffner — APPROVED** [review]
LGTM

**Comment by eunchan** [comment]
As this does not relate to any FPGA tests. I will merge as it is, skipping FPGA test.

**Comment by eunchan** [comment]
image

**Pull request merged**